### PR TITLE
feat: add aggregate_by_weighted_sum to geomapper

### PR DESCRIPTION
**Pull request by dshemetov**
### Description
A function that's being duplicated in nwss and nssp indicators. 

### Changelog
- update geomap docs and lint
- add new aggregation function to geomapper

### CODE REVIEW

File: _delphi_utils_python/data_proc/geomap/README.md [hunk 1]
```diff
@@ -7,42 +7,37 @@ Authors: Jingjing Tang, James Sharpnack, Dmitry Shemetov
...
 - ZIP -> HRR -> HSA crosswalk file comes from the 2018 version at the [Dartmouth Atlas Project](https://atlasdata.dartmouth.edu/static/supp_research_data).
 - FIPS -> MSA crosswalk file comes from the September 2018 version of the delineation files at the [US Census Bureau](https://www.census.gov/geographies/reference-files/time-series/demo/metro-micro/delineation-files.html).
-- State Code -> State ID -> State Name comes from the ANSI standard at the [US Census](https://www.census.gov/library/reference/code-lists/ansi.html#par_textimage_3). The first two digits of a FIPS codes should match the state code here.
+- State Code -> State ID -> State Name comes from the ANSI standard at the [US Census](https://www.census.gov/library/reference/code-lists/ansi.html#par_textimage_3).
```
Comment (dsweber2) [inline _delphi_utils_python/data_proc/geomap/README.md:34]:
```suggestion
- State Code -> State ID -> State Name comes from the ANSI standard at the [US Census](https://www.census.gov/library/reference/code-lists/ansi.html#par_textimage_3). The first two digits of a FIPS codes should match the state code here.
```

File: _delphi_utils_python/delphi_utils/geomap.py [hunk 2]
```diff
@@ -18,54 +18,89 @@ class GeoMapper:  # pylint: disable=too-many-public-methods
...
+    - hhs:          an integer from 1-10, also known as health and human services region
+                    https://www.hhs.gov/about/agencies/iea/regional-offices/index.html
+
+    Valid mappings:
+
+    From            To              Population Weighted
+    zip             fips            Yes
+    zip             hrr             No
+    zip             msa             Yes
+    zip             state_*         Yes
```
Comment (dsweber2) [inline _delphi_utils_python/delphi_utils/geomap.py:43]: Why is it `state_*`?

**Review by dsweber2 — COMMENTED** [review]
I would probably also transfer over some of the tests to make sure this function is working as intended.

Also, someone who is not me should probably check the docs.

File: _delphi_utils_python/delphi_utils/geomap.py (hunk 2 above)
Comment (dshemetov) [reply _delphi_utils_python/delphi_utils/geomap.py:43]: Ah yea that's my referent for `state_id, state_code, state_name`, without having to type all those out. I can add an explainer for that above.

File: _delphi_utils_python/delphi_utils/geomap.py [hunk 3]
```diff
@@ -18,54 +18,89 @@ class GeoMapper:  # pylint: disable=too-many-public-methods
...
+    - zip:          five characters [0-9] with leading 0's, e.g. "33626"
+                    also known as zip5 or zip code
+    - fips:         five characters [0-9] with leading 0's, e.g. "12057"
+                    the first two digits are the state FIPS code and the last
+                    three are the county FIPS code
+    - msa:          five characters [0-9] with leading 0's, e.g. "90001"
+                    also known as metropolitan statistical area
+    - state_code:   two characters [0-9], e.g "06"
+    - state_id:     two characters [A-Z], e.g "CA"
+    - state_name:   human-readable name, e.g "California"
```
Comment (dshemetov) [inline _delphi_utils_python/delphi_utils/geomap.py:31]:
```suggestion
    - state_name:   human-readable name, e.g "California"
    - state_*:          we use this below to refer to the three above geocodes in aggregate
```

File: _delphi_utils_python/delphi_utils/geomap.py [hunk 4]
```diff
@@ -229,13 +264,6 @@ def add_geocode(
     ):
         """Add a new geocode column to a dataframe.
```
Comment (dshemetov) [inline _delphi_utils_python/delphi_utils/geomap.py:265]:
```suggestion
        """Add a new geocode column to a dataframe.
        
        See class docstring for supported geocode transformations.
```

File: _delphi_utils_python/delphi_utils/geomap.py [hunk 5]
```diff
@@ -345,13 +375,6 @@ def replace_geocode(
     ) -> pd.DataFrame:
         """Replace a geocode column in a dataframe.
```
Comment (dshemetov) [inline _delphi_utils_python/delphi_utils/geomap.py:376]:
```suggestion
        """Replace a geocode column in a dataframe.
        
        See class docstring for supported geocode transformations.
```

File: _delphi_utils_python/delphi_utils/geomap.py [hunk 6]
```diff
@@ -627,3 +651,46 @@ def get_geos_within(
...
+    def aggregate_by_weighted_sum(
+        self, df: pd.DataFrame, to_geo: str, sensor: str, population_column: str
+    ) -> pd.DataFrame:
+        """Aggregate sensor, weighted by time-dependent population.
+
+        Note: This function generates its own population weights and adjusts the
+        weights based on which data is NA. This is in contrast to the
+        `replace_geocode` function, which assumes that the weights are already
+        present in the data and does not adjust for missing data (see the
+        docstring for the GeoMapper class).
```
Comment (dshemetov) [inline _delphi_utils_python/delphi_utils/geomap.py:664]:
```suggestion
        Note: This function generates its own population weights and excludes
        locations where the data is NA, which is effectively an extrapolation assumption 
        to the rest of the geos.  This is in contrast to the `replace_geocode` function,
        which assumes that the weights are already present in the data and does
        not adjust for missing data (see the docstring for the GeoMapper class).
```

**Review by dsweber2 — APPROVED** [review]
tests

**Pull request merged**